### PR TITLE
Lesson_1_5

### PR DESCRIPTION
**Pull request by ElenaGor8**
https://github.com/Code-the-Dream-School/dove-react/wiki/Lesson-1.5
Save Todo List in Storage
Create Custom Hook
Fragments

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,20 +2,33 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList'))
```
Comment (mnca16) [inline src/App.js:7]: If the value returned from locastorage is null (when there aren't items in the list), the app would break - because the map method can't read falsey values (null or undefined).

You can see this error in the console:

error-elena

"Can't read properties of null" In order to prevent your app from breaking when having a falsey value, there is a trick I learned from a mentor in the React class:

value || [] is a quick shorthand for "set to empty array if value is falsey"

in other words:

React.useState(JSON.parse(localStorage.getItem('savedTodoList')) || [])

File: src/App.js [hunk 2]
```diff
@@ -2,20 +2,33 @@ import React from 'react';
...
 function App() {
-  const [todoList, setTodoList] = React.useState([]);
+  const [todoList, setTodoList] = useSemiPersistentState();
 
   function addTodo(newTodo) {
     setTodoList([...todoList, newTodo]);
   }
   return (
-    <div>
+    <>
```
Comment (mnca16) [inline src/App.js:25]: 💯

**Review by mnca16 — COMMENTED** [review]
Excellent work Elena - keep it up! 🥇 I left a comment that might be helpful, let me know if you have questions.
